### DISCUSSION TITLE
Faulted workflow on Elsa Server restart

### DISCUSSION THREAD

**Discussion by taurisjoe** [elsa-workflows/elsa-core#3315] [Q&A]
Is there a way to prevent Elsa Server from retrying faulted workflows on startup?  

I'm finding that I have to manually delete them from the WorkflowInstances table manually or drop the entire workflows database to have Elsa regenerate it and then restore my workflows from json.

**Comment by sfmskywalker** [comment]
I don't think that Elsa retries faulted workflows automatically. However, it will try to continue executing workflows that are in the `Running` state. Is this what you are referring to?

If yes, then right now you would either have to copy the `AddElsa` and `AddElsaCore` methods and update the latter on to *not* add the `ContinueRunningWorkflows` startup-task, or you have to remove this service from the container (both the implementation as the service as well as `IStartupTask` as the service for the implementation).

A better option would be to make this service optional and allow you to control whether this service should be registered through Elsa Options. Please feel free to submit a PR or open an issue for this.
